### PR TITLE
Fixes #14703: Let rtf execute tests from a given ncf PR

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/14703

### CODE REVIEW

File: rtf [hunk 1]
```diff
@@ -115,6 +115,7 @@ Commands
         Setup the given platform, install ncf from git in the specified version and then run the ncf tests.
         When no cfengine version is specified, tests will be executed with the cfengine agent shipped with the
         Rudder agent in the same version than the ncf version specified.
+        You can also test a give ncf pull request by remplacing the version parameter by the redmine number, like #14641
```
Comment (peckpeck) [inline rtf]: beware, # in a command line is a comment, can't we have something else ?

File: scenario/lib.py [hunk 2]
```diff
@@ -231,25 +233,44 @@ def finish():
...
+       - CONTINUE: continue testing even if this fail, should be the default
+       - BREAK: stop the scenario if this fail, for tests that change a state
+       - IGNORE: will ignore the test test result in the global testing result
+  """
   try:
     if live_output:
       print("+" + command)
     if hostname == 'localhost':
-      return check_output("LANG=C " + command, shell=True)
+        return ("0", check_output("LANG=C " + command, shell=True))
```
Comment (peckpeck) [inline scenario/lib.py]: i think the return code should be an integer

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
Commit modified

File: rtf [hunk 3]
```diff
@@ -22,6 +22,7 @@ Usage:
     rtf scenario doc <scenario>
     rtf scenario env <platform>
     rtf scenario ncf_tests <platform> <version> [--cfengine-version=<cfengine_version>]
+    rtf scenario ncf_tests <platform> <pull-request-number> [--cfengine-version=<cfengine_version>]
```
Comment (peckpeck) [inline rtf]: this won't work

**Pull request merged**